mne/io/edf/tests/test_edf.py

```diff
@@ -125,10 +125,14 @@ def test_edf_data():
         read_raw_edf(broken_fname, exclude=raw.ch_names[:132], preload=True)
 
 
-@pytest.mark.filterwarnings('ignore::RuntimeWarning')
 def test_duplicate_channel_labels_edf():
     """Test reading edf file with duplicate channel names."""
-    raw = read_raw_edf(duplicate_path, preload=True)
+    # reading files with dupiicate names should generate a runtime warning
+    with pytest.warns(RuntimeWarning, match='Channel names are not unique'):
+        raw = read_raw_edf(duplicate_path, preload=True)
+        # This file contains the channel name 'EEG F1-Ref' twice
+        assert 'EEG F1-Ref-0' in raw.ch_names
```

Comment:
The assert should be outside of the pytest context.

I would actually check all the channels in a single go.

ie:
```py
def test_duplicate_channel_labels_edf():
    """Test reading edf file with duplicate channel names."""
    EXPECTED_CHANNEL_NAMES = ['EEG F1-Ref-0', 'EEG F2-Ref', 'EEG F1-Ref-1']
    with pytest.warns(RuntimeWarning, match='Channel names are not unique'):
        raw = read_raw_edf(edf_with_duplicated_channels_path, preload=False)
    assert all(raw.ch_names == EXPECTED_CHANNEL_NAMES)
```

(plus) `preload=False` since you don't need to load the data to test that